File: from-1001-to-1100/1002-find-common-characters/solution.go
func commonChars(words []string) []string {
    counters := make([]map[byte]int, len(words))
    var cnt map[byte]int
    for i, word := range words {
        cnt = make(map[byte]int)
        for i := 0; i < len(word); i ++ {
            cnt[word[i]] += 1
        }
        counters[i] = cnt
    }

    result := []string{}
    for char, numChars := range counters[0] {
        minChars := numChars
        for _, cnt := range counters[1:] {
            num := cnt[char]
            if num < minChars {
                minChars = num
            }
            if minChars == 0 {
                break
            }
        }
        for i := 0; i < minChars; i ++ {
            result = append(result, string(char))
        }
    }
    return result
}

File: from-1001-to-1100/1002-find-common-characters/solution.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        counters = [Counter(list(word)) for word in words]
        
        ans = []
        for k in counters[0].keys():
            if all([k in c for c in counters[1:]]):
                ans.extend([k] * min([c[k] for c in counters]))
                
        return ans
    